feat: label un-audited challenges

ojeytonwilliams opened the pull request:
This shows how we could control this at the certificate level.  If you set the `CURRICULUM_LOCALE=espanol` and go to learn/responsive-web-design/ it should not be visible, but it should be on learn/front-end-libraries/

ojeytonwilliams committed:
feat: label un-audited challenges

nhcarrigan commented:
Do we want this to display on every challenge?

Or on the superblock buttons on the map?

raisedadead commented:
We would ideally display:

Landing page: No change whatsoever.
/learn: A horizontal separation between challenges ready and those pending translation, with a tiny CTA link to translate.
/learn/super-block: One pill per course (aka block).
/learn/super-block/block/challenge: Somewhere with a tiny CTA link to translate.

Commit:
feat(client): split learn map

Splits the learn map into two sections - one for translated superBlocks,
the other for untranslated. Adds small CTA link after untranslated
section.

nhcarrigan commented:
![image](https://user-images.githubusercontent.com/63889819/107320606-e4e66180-6a55-11eb-9211-6ca9fa21ae08.png)

`/learn` map separated by translation status.

ojeytonwilliams commented on client/src/components/Map/index.js:
```diff
@@ -69,6 +71,51 @@ function renderLearnMap(nodes, currentSuperBlock = '') {
+            </Link>
+          </li>
+        ))}
+      <hr />
```
With the CTA at the bottom, I don't think it's clear what this `hr` indicates.  Could we put the text here?

For the text itself, how about

"The following certificates are still being translated.  Please go here(link to contribute docs) if you think you can help."

raisedadead replied:
I agree, we should move that up.

nhcarrigan replied:
I'll get this moved :)

ojeytonwilliams commented:
@nhcarrigan I'm converting my bit from `TRANSLATION PENDING` to pills.

nhcarrigan commented:
> @nhcarrigan I'm converting my bit from TRANSLATION PENDING to pills.

Sounds good. I had a draft started for pills, but finally got frustrated enough with Windows and moved my dev environment to WSL. (Without remembering to push up that draft, of course 😅 )

ojeytonwilliams commented:
Current plan for style of CTAs/warnings:

- /learn/super-block: link to contribution docs (with the words 'help translate').
- /learn/super-block/block/challenge: subtle (info colour) banner (again 'help translate')

ojeytonwilliams commented:
> Sounds good. I had a draft started for pills, but finally got frustrated enough with Windows and moved my dev environment to WSL. (Without remembering to push up that draft, of course  )

Ouch... I hope you didn't lose too much.

I'll push up those changes ASAP so you can play around with them.  I'll not touch what you've already worked on, just the two paths I mentioned above.

ojeytonwilliams committed:
feat: add translation CTA to challenges

ojeytonwilliams commented:
![image](https://user-images.githubusercontent.com/15801806/107400100-cb9aef00-6b01-11eb-81a3-df4b29fa7479.png)

The 'ES' is just for debugging.

nhcarrigan commented:
Hmm - I think it'd be better under the title, rather than next to it. 🤔

ojeytonwilliams commented:
I honestly can't decide.  I've tried above, below and to the right, but none of them looked great.  To the right looked least bad to me, but not by much.

ojeytonwilliams committed:
refactor: updated comment about cert extensions

nhcarrigan commented:
![image](https://user-images.githubusercontent.com/63889819/107402913-5c011180-6ab9-11eb-9ba8-a1c84c18677c.png)

Better `/learn` CTA

Commit:
fix: move learn CTA

Moves the learn CTA up to the top of the non-translated cert
section. Also fixes the linting issue from previous commit.

ojeytonwilliams committed:
fix: revert map's per challenge CTAs

ojeytonwilliams committed:
feat: add cta links to blocks

ojeytonwilliams commented:
![image](https://user-images.githubusercontent.com/15801806/107408939-e83c2480-6b0b-11eb-81e2-c6089029047c.png)

How this looks on http://localhost:8000/learn/javascript-algorithms-and-data-structures/

nhcarrigan commented:
![image](https://user-images.githubusercontent.com/63889819/107409338-e8fb9900-6ac0-11eb-87b4-85cd9522f1ff.png)

Thoughts on rounding it off to look more like a pill?

nhcarrigan commented:
![image](https://user-images.githubusercontent.com/63889819/107409444-07fa2b00-6ac1-11eb-9633-5e8659381ffd.png)

Also wondering if we should set a fixed width

ojeytonwilliams commented:
I'm not sure about the pills, since the site's aesthetic is quite sharp.  Aside from a few circles (check marks and test icons) it's pretty much all straight lines.

The text shouldn't wrap, though, I agree. I've pushed a commit to stop that element shrinking.

nhcarrigan commented:
I'll give this a rebase to resolve the conflict.

Commit:
fix(tools): update snapshot for jest

Updates the snapshot to reflect the addition of the translation
pill.

nhcarrigan commented:
@ojeytonwilliams I think we've covered everything now - is this ready for review?

raisedadead commented:
Quick rebase for the QA please?

ojeytonwilliams commented:
Just resolving that now.

raisedadead committed:
fix(client): remove translation banner

raisedadead commented:
Here is one more take:

![image](https://user-images.githubusercontent.com/1884376/107543663-6d4c3a00-6bef-11eb-86d7-1eb9d8acac75.png)


![image](https://user-images.githubusercontent.com/1884376/107543684-72a98480-6bef-11eb-86df-3bc56effd827.png)


![image](https://user-images.githubusercontent.com/1884376/107543707-79d09280-6bef-11eb-9e34-f1951a830efa.png)

![image](https://user-images.githubusercontent.com/1884376/107543741-8359fa80-6bef-11eb-9d0e-6f375b3ac9e2.png)

raisedadead committed:
fix(client): add spacer between CTA and cert btn

raisedadead committed:
fix(client): update styles for block and title CTA

raisedadead commented:
I added a spacer between the CTA on the `/learn`. Let me know if you would like me to add the other styles for the block and title CTA in the screenshot above.

nhcarrigan commented:
Ooo I really like those. I'm 100% for it.

ojeytonwilliams commented:
> Let me know if you would like me to add the other styles for the block and title CTA in the screenshot above.

If it's good to go, yeah, push it up.

nhcarrigan commented:
Hey, with 685e2db included here, landing this should mean we're good to go for the Crowdin uploads again right?

raisedadead commented:
I edited the issue comments in this PR to collapse the previous screenshots (just to make it easy to catch the current status of the PR).

> Hey, with 685e2db included here, landing this should mean we're good to go for the Crowdin uploads again right?

Yes - after this has been through the QA.

RandellDawson reviewed (dismissed):
I tested this locally and it seems to achieve the desired goal.  I kind of wish the labels were all the same height to avoid seeing blue boxes of various heights (due to the text inside the box).  However, these are temporary and will slowly come off as curriculum sections are translated.

ahmadabdolsaheb commented:
> I tested this locally and it seems to achieve the desired goal. I kind of wish the labels were all the same height to avoid seeing blue boxes of various heights (due to the text inside the box). However, these are temporary and will slowly come off as curriculum sections are translated.
working on those :)

ahmaxed committed:
fix: elem hover and positioning

ahmadabdolsaheb commented:
@ojeytonwilliams great work. @raisedadead great design. I just changed alignment, some margin, and replaced the hover state.

The reason we are doing underline for the breadcrumbs is that @nhcarrigan and I have hid a triangle in there that does not play well with hover. We should probably combine it with the left breadcrumb and overlap the right side so each section can be styled onhover properly.

Finally the highlight colors seem too saturated that they steal the show, I will circle back to them and tone them down a little bit.

LGMT.

raisedadead commented:
I am going to update the messages that we agreed upon in the chat in a bit and land this.

raisedadead committed:
fix(i18n): update validated copy of strings

raisedadead reviewed (dismissed):
I would like someone else to approve as well because that last commit is mine.

ojeytonwilliams commented:
I can review it - it's mostly not my work now.

ojeytonwilliams commented on client/src/templates/Introduction/intro.css:
```diff
@@ -8,11 +8,40 @@
+.block-cta-wrapper {
+  display: flex;
+  flex-direction: column;
+  justify-content: start;
+  padding-right: 15px;
+  padding-top: 25px;
```
```suggestion
  justify-content: center;
  padding-right: 15px;
```
@ahmadabdolsaheb couple of things: I got a warning about `start` from Gatsby and don't we just want it to be centered? It looks better to me centered.

ahmadabdolsaheb replied:
Sure, we could either place it in accordance with card or the title. Both are alright with me.

ahmadabdolsaheb replied:
I will give our Accent Colors a round of audit. That could come in another pr.

ojeytonwilliams reviewed (commented):
The night mode colours are a bit ugly
<details>

![image](https://user-images.githubusercontent.com/15801806/107618882-918d3280-6c52-11eb-90bd-501e690e1ee8.png)

![image](https://user-images.githubusercontent.com/15801806/107619655-ca79d700-6c53-11eb-9db5-a0a420a06b45.png)
</details>


something like this looks better to me:

<details>

![image](https://user-images.githubusercontent.com/15801806/107619157-06f90300-6c53-11eb-8278-8f43d2b3b94f.png)

</details>

@ahmadabdolsaheb were you planning on revisiting them, checking contrast and so on?  If so, we can go ahead with this PR.
 
Apparently I can't request changes on my own PR for... reasons, but still:

ahmaxed committed:
feat:center button

ahmadabdolsaheb commented:
Seems like 'misc.translation-pending' and 'learn.help-translate-link' could be combined since both are the same string: "Help us translate.". But it could stay for now.

ahmadabdolsaheb commented on client/i18n/translations-schema.js:
```diff
@@ -448,7 +448,7 @@ const translationsSchema = {
     and: 'and',
     'change-theme': 'Sign in to change theme.',
-    'translation-pending': 'Help translate'
+    'translation-pending': 'Help us translate.'
```
some translation-pending have . and some don't. do we need the or not?

raisedadead replied:
The reason is one is a sentence, as in a link in a sentence. And the other is a button label. We could re-visit this once its out.

raisedadead commented:
> Seems like 'misc.translation-pending' and 'learn.help-translate-link' could be combined since both are the same string: "Help us translate.". But it could stay for now.

Yes, I wanted to consolidate both but kept them separate because one appears with a "stop" symbol on the `/learn` and the other is a button label.

We could re-visit it. 

I think this is good to go.

raisedadead commented on client/i18n/translations-schema.js:
```diff
@@ -445,7 +447,8 @@ const translationsSchema = {
     and: 'and',
-    'change-theme': 'Sign in to change theme.'
+    'change-theme': 'Sign in to change theme.',
+    'translation-pending': 'Help us translate.'
```
```suggestion
    'translation-pending': 'Help us translate'
```

@ahmadabdolsaheb you are correct. Fixing this.

raisedadead committed:
fix: remove additional dot from the schema

raisedadead reviewed (approved):
For the final time.

ojeytonwilliams commented:
I can't technically can't approve, but I do approve.
